frontend/packages/insights-plugin/__tests__/insightswidget.spec.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+import { InsightsPopup } from '../src/components/InsightsPopup/index';
+import fixtures, { funcs } from './insights.fixtures';
+
+describe('InsightsPopup ', () => {
+  const functions = funcs();
+  const props = {
+    ...functions,
+    ...fixtures,
+    responses: { ...fixtures.propsForInsightsWidget.responses },
+    k8sResult: { ...fixtures.propsForInsightsWidget.k8sResult },
+  };
+
+  const wrapper = mount(<InsightsPopup {...props} />);
+  it('main container renders correctly', () => {
+    expect(wrapper.debug()).toMatchSnapshot();
+  });
+  it('should have the correct error component', () => {
+    expect(wrapper.find('.co-insights__box'));
+  });
+  it('should have correct text inside co-insights__box', () => {
+    expect(wrapper.find('.insights-widget-text-enabled').text()).toBe(
+      'Insights Advisor identifies and prioritizes risks to security, performance, availability, and stability of your clusters',
+    );
+  });
+});
+
+describe('InsightsPopup error state', () => {
+  const functions = funcs();
+  const props = {
```

Comment:
Make sure, you have to modify responses, so you can do something like
const responses = fixtures.responses; responses[0].response.data.result = [];

Reply:
then you can do 
```suggestion
  const props = {
     ...fixtures,
     responses
  }
```